File: backend/src/domain/scheduling/schema.rs
#![allow(dead_code)]
use serde::{Deserialize, Serialize};

// Scheduling domain schemas

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Scheduler {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub cron_expression: String,
    pub enabled: Option<bool>,
    pub next_run_at: Option<String>,
    pub last_run_at: Option<String>,
    pub last_run_duration: Option<i64>,
    pub last_run_status: Option<String>,
    pub last_run_error: Option<String>,
    pub last_run_result: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct CronExpression {
    pub id: i64,
    pub name: String,
    pub description: Option<String>,
    pub expression: String,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct ScheduledTask {
    pub id: i64,
    pub scheduler_id: i64,
    pub task_name: String,
    pub task_type: String,
    pub task_config: Option<String>,
    pub status: Option<String>,
    pub error: Option<String>,
    pub created_at: Option<String>,
    pub started_at: Option<String>,
    pub completed_at: Option<String>,
    pub next_run_at: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetSchedulerReq {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetSchedulerRes {
    pub scheduler: Scheduler,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchSchedulersReq {
    pub name: Option<String>,
    pub enabled: Option<bool>,
    pub page: i64,
    pub limit: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchSchedulersRes {
    pub schedulers: Vec<Scheduler>,
    pub total: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddSchedulerReq {
    pub name: String,
    pub description: Option<String>,
    pub cron_expression: String,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddSchedulerRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdateSchedulerReq {
    pub id: i64,
    pub name: Option<String>,
    pub description: Option<String>,
    pub cron_expression: Option<String>,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdateSchedulerRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetCronExpressionReq {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetCronExpressionRes {
    pub cron_expression: CronExpression,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchCronExpressionsReq {
    pub name: Option<String>,
    pub enabled: Option<bool>,
    pub page: i64,
    pub limit: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchCronExpressionsRes {
    pub cron_expressions: Vec<CronExpression>,
    pub total: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddCronExpressionReq {
    pub name: String,
    pub description: Option<String>,
    pub expression: String,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddCronExpressionRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdateCronExpressionReq {
    pub id: i64,
    pub name: Option<String>,
    pub description: Option<String>,
    pub expression: Option<String>,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdateCronExpressionRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetScheduledTaskReq {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct GetScheduledTaskRes {
    pub scheduled_task: ScheduledTask,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchScheduledTasksReq {
    pub scheduler_id: Option<i64>,
    pub task_type: Option<String>,
    pub status: Option<String>,
    pub page: i64,
    pub limit: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct SearchScheduledTasksRes {
    pub scheduled_tasks: Vec<ScheduledTask>,
    pub total: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddScheduledTaskReq {
    pub scheduler_id: i64,
    pub task_name: String,
    pub task_type: String,
    pub task_config: Option<String>,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct AddScheduledTaskRes {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdateScheduledTaskReq {
    pub id: i64,
    pub scheduler_id: i64,
    pub task_name: Option<String>,
    pub task_type: Option<String>,
    pub task_config: Option<String>,
    pub enabled: Option<bool>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct UpdateScheduledTaskRes {
    pub id: i64,
}

File: backend/domain/scheduling/schema.go
package scheduling

import "time"

type Task struct {
	ID             int64      `json:"id"`
	Name           string     `json:"name"`
	Description    string     `json:"description"`
	TaskType       string     `json:"task_type"`
	SchedulerID    int64      `json:"scheduler_id"`
	CronExpression string     `json:"cron_expression"`
	Enabled        bool       `json:"enabled"`
	NextRunAt      *time.Time `json:"next_run_at"`
	LastRunAt      *time.Time `json:"last_run_at"`
	LastRunStatus  string     `json:"last_run_status"`
	LastRunError   string     `json:"last_run_error"`
	CreatedAt      *time.Time `json:"created_at"`
	UpdatedAt      *time.Time `json:"updated_at"`
}

type GetTaskReq struct {
	ID int64 `json:"id" binding:"required"`
}

type SearchTasksReq struct {
	Name        string `json:"name"`
	TaskType    string `json:"task_type"`
	SchedulerID int64  `json:"scheduler_id"`
	Enabled     *bool  `json:"enabled"`
	Page        int64  `json:"page"`
	Limit       int64  `json:"limit"`
}

type AddTaskReq struct {
	Name           string `json:"name" binding:"required"`
	TaskType       string `json:"task_type" binding:"required"`
	SchedulerID    int64  `json:"scheduler_id" binding:"required"`
	CronExpression string `json:"cron_expression"`
	Enabled        *bool  `json:"enabled"`
}

type UpdateTaskReq struct {
	ID             int64   `json:"id" binding:"required"`
	Name           *string `json:"name"`
	TaskType       *string `json:"task_type"`
	SchedulerID    *int64  `json:"scheduler_id"`
	CronExpression *string `json:"cron_expression"`
	Enabled        *bool   `json:"enabled"`
}

type RunTaskReq struct {
	ID int64 `json:"id" binding:"required"`
}

type StopTaskReq struct {
	ID int64 `json:"id" binding:"required"`
}

type RestartTaskReq struct {
	ID int64 `json:"id" binding:"required"`
}

type GetTaskRes struct {
	Task *Task `json:"task"`
}

type SearchTasksRes struct {
	Tasks []*Task `json:"tasks"`
	Total int64  `json:"total"`
}

type AddTaskRes struct {
	ID int64 `json:"id"`
}

type UpdateTaskRes struct {
	ID int64 `json:"id"`
}

type TaskActionRes struct {
	ID int64 `json:"id"`
}
